wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,259 @@
+
+## Usage Example
+
+```js
+const { BrowserView, BrowserWindow, ContainerView, ScrollView, WrapperBrowserView } = require("electron");
+
+const window = new BrowserWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new ContainerView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const browserView1 = new BrowserView()
+const wrapperBrowserView1 = new WrapperBrowserView({ 'browserView': browserView1 });
+wrapperBrowserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addBrowserView(wrapperBrowserView1)
+
+const browserView2 = new BrowserView()
+const wrapperBrowserView2 = new WrapperBrowserView({ 'browserView': browserView2 });
+wrapperBrowserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addBrowserView(wrapperBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChildView(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+## BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+### Static Methods
+
+#### `BaseView.getAllViews()`
+
+Returns the array of all created views.
+
+#### `BaseView.fromId(id)`
+
+Returns `BaseView | null` - The view with the given `id`.
+
+### Instance Properties
+
+#### `view.id`
+
+A `Integer` property representing the unique ID of the view.
+
+#### `view.isContainer`
+
+A `boolean` property that determines whether this view is `ContainerView`.
+
+### Instance Methods
+
+#### `view.setBounds(bounds)`
+
+Resizes and moves the view to the supplied bounds relative to its parent.
+
+#### `view.getBounds()`
+
+Returns the position and size of the view.
+
+#### `view.setBackgroundColor(color)`
+
+Change the background color of the view.
+
+#### `view.getParentView()`
+
+Returns `BaseView || null` - The parent view, otherwise returns `null`.
+
+#### `view.getParentWindow()`
+
+Returns `BrowserWindow || null` - The window that the view belongs to, otherwise returns `null`.
+
+### Events
+
+#### Event: 'size-changed'
+
+Emitted when the view's size has been changed.
+
+ 
+## BrowserWindow 
+
+### New instance methods
+
+#### `win.addChildView(view)`
+
+#### `win.removeChildView(view)`
+
+#### `win.getViews()`
+
+Returns `BaseView[]` - an array of all BaseViews that have been attached
+with `addChildView`.
+
+
+## ContainerView
+
+A `ContainerView` can be used to embed additional views hierarchy into a `BrowserWindow`. It extends `BaseView`.
+
+Process: **Main**
+
+### Instance Methods
+
+#### `containerView.addChildView(view)`
+
+#### `containerView.removeChildView(view)`
+
+#### `containerView.getViews()`
+
+Returns `BaseView[]` - an array of all BaseViews that have been attached
+with `addChildView`.
```

Comment:
nit: I feel like these should just be `containerView.addChild(view)` / `containerView.removeChild(view)` / `containerView.children`. No need to repeat the word `view` over and over.

Reply:
I agree

Reply:
I renamed getViews on getChildren - as I don't know what to do with BaseWindow.getChildren/getViews; I feel that it shouldn't be a property in such case (because what about BaseWindow.getBrowserViews?...)

Reply:
Restored the name `8418e0413a4cccc2258add477052713d647a9300` - currently defined in `View`.